nemo/collections/asr/parts/utils/online_clustering.py

```diff
@@ -1189,3 +1196,422 @@ def forward_infer(
         # Match the permutation of the newly obtained speaker labels and the previous labels
         merged_clus_labels = self.match_labels(Y_merged=Y, add_new=add_new)
         return merged_clus_labels
+
+
+class LongFormSpeakerClustering(SpeakerClustering):
+    def __init__(
+        self, unit_window_len: int = 10000, sub_cluster_n: int = 50, cuda: bool = False,
+    ):
+        """
+        Initializes the speaker clustering class tailored for long-form audio, inheriting from the `SpeakerClustering` class.
+        The algorithm for long-form clustering is executed through the `forward_infer` function. Input embedding vectors are 
+        divided into smaller windows, each of size `unit_window_len`. Within every window, the clustering algorithm strives 
+        to identify `sub_cluster_n` distinct clusters. The obtained clustering labels are then expanded to match the original 
+        length of the input embeddings.
+
+        Args:
+            sub_cluster_n (int):
+                The target number of clusters to identify within each window.
+            unit_window_len (int):
+                The size of the windows in which the algorithm aims to identify `sub_cluster_n` clusters.
+            cuda (bool):
+                Boolean variable for toggling cuda availability.
+            device (torch.device):
+                Specifies the device for computation, valid values are `torch.device("cpu")`, `torch.device("cuda:<int>")`, or 
+                `torch.device("cuda")`.
+        """
+        super().__init__()
+        self.speaker_clustering = SpeakerClustering(cuda=cuda)
```

Comment:
Moved `LongFormSpeakerClustering` to longform_clustering.py.
Now `LongFormSpeakerClustering` is not inheriting any class, but initializing `SpeakerClustering` in it to use its methods (because torch.jit.script does not support inherited methods)